File: app/src/main/java/com/example/woo/songstar/utils/AppSharedPreferences.java
package com.example.woo.songstar.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class AppSharedPreferences {
    public static final String USER_ID = "user_id";
    public static final String NAME = "name";
    public static final String USERNAME = "userName";
    public static final String LANGUAGE = "language";
    public static final String PHOTO = "photo";
    public static final String IS_ADMIN = "isAdmin";


    private static AppSharedPreferences appSharedPreferences;
    public static AppSharedPreferences getInstance() {
        if(appSharedPreferences == null){
            appSharedPreferences = new AppSharedPreferences();
        }

        return appSharedPreferences;
    }

    public void putString(Context context, String key, String value) {
        try{
            SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.putString(key, value);
            editor.apply();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public String getString(Context context, String key) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getString(key, null);
    }

    public void putBoolean(Context context, String key, boolean value) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public boolean getBoolean(Context context, String key) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getBoolean(key, false);
    }

    public void clearAllPreferences(Context context) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.clear();
        editor.apply();
    }
}

File: app/src/main/java/com/example/woo/songstar/utils/AppSharedPreferences.kt
package com.example.woo.songstar.utils

import android.content.Context
import androidx.preference.PreferenceManager

object AppSharedPreferences {
    const val USER_ID = "user_id"
    const val NAME = "name"
    const val USERNAME = "userName"
    const val LANGUAGE = "language"
    const val PHOTO = "photo"
    const val IS_ADMIN = "isAdmin"

    fun putString(context: Context?, key: String?, value: String?) {
        try {
            val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
            val editor = sharedPreferences.edit()
            editor.putString(key, value)
            editor.apply()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun getString(context: Context?, key: String?): String? {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        return sharedPreferences.getString(key, null)
    }

    fun putBoolean(context: Context?, key: String?, value: Boolean) {
        val sharedPref = PreferenceManager.getDefaultSharedPreferences(context)
        val editor = sharedPref.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getBoolean(context: Context?, key: String?): Boolean {
        val sharedPref = PreferenceManager.getDefaultSharedPreferences(context)
        return sharedPref.getBoolean(key, false)
    }

    fun clearAllPreferences(context: Context?) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val editor = sharedPreferences.edit()
        editor.clear()
        editor.apply()
    }

}